internal/kuberesource/parts.go

```diff
@@ -553,36 +553,44 @@ func ServiceMeshProxy() *applycorev1.ContainerApplyConfiguration {
 func CryptsetupInitCommand() string {
 	return `#!/bin/bash
 set -e
+
+# Regarding https://man7.org/linux/man-pages/man8/cryptsetup.8.html
+# passphrase <=> key in this document, not an actual cryptographic key
+
 # device is the path to the block device to be encrypted.
 device="/dev/csi0"
 # workload_secret_path is the path to the Contrast workload secret.
 workload_secret_path="/contrast/secrets/workload-secret-seed"
+
 # tmp_key_path is the path to a temporary key file.
 tmp_key_path="/dev/shm/key"
 # disk_encryption_key_path is the path to the disk encryption key.
 disk_encryption_key_path="/dev/shm/disk-key"
 
-# If the device is not already a LUKS device, format it.
+# (First initialization)
 if ! cryptsetup isLuks "${device}"; then
-	# Generate a random key for the first initialization.
-	dd if=/dev/urandom bs=1 count=32 2>/dev/null | base64 -w0 > "${tmp_key_path}"
+	# First init requires formatting to LUKS to allow getting the uuid of LUKS block device.
+	echo "init_passphrase" > "${tmp_key_path}"
 	cryptsetup luksFormat --pbkdf-memory=10240 $device "${tmp_key_path}" </dev/null
 
-	uuid=$(blkid "${device}" -s UUID -o value)
-	openssl kdf -keylen 32 -kdfopt digest:SHA2-256 -kdfopt hexkey:$(cat "${workload_secret_path}") \
-		-kdfopt info:${uuid} -binary HKDF | base64 -w0 > "${disk_encryption_key_path}" 2>/dev/null
+	# Generate passphrase based on workload secret and uuid of LUKS device.
+	echo "$(cat "${workload_secret_path}")$(cryptsetup luksUUID "${device}")" > "${disk_encryption_key_path}"
```

Comment:
```suggestion
	cat "${workload_secret_path}" > "${disk_encryption_key_path}"
	cryptsetup luksUUID "${device}" >> "${disk_encryption_key_path}"
```
... is a bit easier to read and does not need to care about `echo` peculiarities. Applies to the other if branch, too.